zkabacus-crypto/src/states.rs

```diff
@@ -191,10 +205,16 @@ impl MerchantBalance {
     pub fn into_inner(self) -> u64 {
         self.0.into_inner()
     }
+
+    /// Try to add the value of the given [`CustomerBalance`] to self.
+    /// Fails if the sum is too large.
+    pub fn try_add(self, rhs: CustomerBalance) -> Result<Self, Error> {
```

Comment:
I understand if we only need this specific functionality, but why don't we have the same functionality on `CustomerBalance`? Or a `try_sub()`? What is the use case actually, getting the total amount on the channel? Do we have these for adding a normal `Balance` to both MerchantBalance and CustomerBalance?

Reply:
This is used for scenarios where the customer balance is transferred to the merchant e.g. a successful expiry or a successful dispute. There is no scenario in the protocol where we subtract balances or transfer the `MerchantBalance` to the customer---every other transfer of money is done in a payment and the balances change by a `PaymentAmount`.